packages/autoprofile/precompile/build-single-addon.js

```diff
@@ -29,8 +30,17 @@ module.exports = exports = function buildSingleAddOn(abi, version) {
     mkdirpSync(addonDir);
   }
 
-  execSync(`node node_modules/node-gyp/bin/node-gyp.js rebuild --target=${version} --arch=x64`, {
-    cwd: path.join(__dirname, '..'),
+  let darwinNodeJs10Fix = '';
+  if (platform === 'darwin' && semver.gte(version, '10.0.0') && semver.lt(version, '12.0.0')) {
+    darwinNodeJs10Fix = '--build_v8_with_gn=false';
```

Comment:
It failed with an error without this extra option. Something like this:

```
**gyp: name 'build_v8_with_gn' is not defined while evaluating condition 'build_v8_with_gn == "true"' in binding.gyp while trying to load binding.gyp**
gyp ERR! configure error 
```

I took the workaround from here: https://github.com/nodejs/node-gyp/issues/1457#issuecomment-395490546

Since we no longer publish prebuild binaries for Darwin (this was also changed in this PR) it probably does not matter much. The Darwin binaries are only used for local testing now.